test/test_tensorexpr_pybind.py

```diff
@@ -67,31 +67,6 @@ def test_external_calls(self):
         codegen.call([tA, tB, tC])
         torch.testing.assert_close(torch.matmul(tA, tB), tC)
 
-    def test_dynamic_shape(self):
-        dN = te.VarHandle(torch.int32)
-        A = te.BufHandle(torch.float64)
-        B = te.BufHandle(torch.float64)
-
-        def compute(i):
-            return A.load(i) - B.load(i)
-
-        C = te.Compute("C", [dN], compute)
-
-        loopnest = te.LoopNest([C])
-        loopnest.prepare_for_codegen()
-
-        cg = te.construct_codegen("ir_eval", loopnest.simplify(), [A, B, C, dN])
-
-        def test_with_shape(n):
-            tA = torch.randn(n, dtype=torch.double)
-            tB = torch.randn(n, dtype=torch.double)
-            tC = torch.empty(n, dtype=torch.double)
-            cg.call([tA, tB, tC, n])
-            torch.testing.assert_close(tA - tB, tC)
-
-        test_with_shape(8)
-        test_with_shape(31)
-
```

Comment:
@huiguoo @ZolotukhinM Do you think it's reasonable to remove this test? Or should we use an alternative solution in order to keep this test, e.g. adding a special case for {dims.size() == 0 && indices.size() == 1}

Context: this test creates `BufHandle`s with 0 dimensions and then tests on inputs of different sizes, so it's incompatible with out-of-bounds checks. Note that the 2d version of this test already fails, due to a check in `flatten_index` (since if `indices.size() > strides.size()`, we can't compute the flattened index)

Reply:
I think we can fix the test:
```diff
diff --git a/test/test_tensorexpr_pybind.py b/test/test_tensorexpr_pybind.py
index 2a0dbfe293a..09d93f8f711 100644
--- a/test/test_tensorexpr_pybind.py
+++ b/test/test_tensorexpr_pybind.py
@@ -69,8 +69,8 @@ class TestTensorExprPyBind(JitTestCase):

     def test_dynamic_shape(self):
         dN = te.VarHandle(torch.int32)
-        A = te.BufHandle(torch.float64)
-        B = te.BufHandle(torch.float64)
+        A = te.BufHandle("A", [dN], torch.float64)
+        B = te.BufHandle("B", [dN], torch.float64)

         def compute(i):
             return A.load(i) - B.load(i)
```
Will it pass after this edit?

Reply:
> Note that the 2d version of this test already fails

Where is the 2d version?

Reply:
This is the test I wrote for 2D:

```python
    def test_dynamic_shape_2d(self):
        zero = te.ExprHandle.int(0)
        dM = te.VarHandle(torch.int32)
        dN = te.VarHandle(torch.int32)
        i = te.VarHandle(torch.int32)
        j = te.VarHandle(torch.int32)
        A = te.BufHandle(torch.float64)
        B = te.BufHandle(torch.float64)
        C = te.BufHandle(torch.float64)

        a_val = A.load([i, j])
        b_val = B.load([i, j])
        add_val = a_val + b_val
        body = te.Store.make(C, [i, j], add_val)

        inner = te.For.make(j, zero, dN, body)
        stmt = te.For.make(i, zero, dM, inner)

        cg = te.construct_codegen("ir_eval", stmt, [A, B, C, dN, dM])

        def test_with_shape(n, m):
            tA = torch.randn(n, m, dtype=torch.double)
            tB = torch.randn(n, m, dtype=torch.double)
            tC = torch.empty(n, m, dtype=torch.double)
            cg.call([tA, tB, tC, n, m])
            torch.testing.assert_close(tA + tB, tC)

        test_with_shape(2, 4)
        test_with_shape(6, 3)
```

Reply:
Yeah, we just need to specify dimensions for the buffers we create ([dN, dM]), you can do that in the 2d version as well. Previously the test was not rigorous enough, but we got away with that because we didn't have the out-of-bounds check. Now we need to be more careful and consistent with how we define buffers.

Also, you don't need to create IR manually, you could use the same `Compute` API as in 1-d case - this would probably let you make the test slightly more concise.